test/parallel/test-http-outgoing-message-inheritance.js

```diff
@@ -0,0 +1,32 @@
+'use strict';
+
+const common = require('../common');
+const { OutgoingMessage } = require('http');
+const { Writable } = require('stream');
+const assert = require('assert');
+
+// check that OutgoingMessage can be used without a proper Socket
+// Fixes: https://github.com/nodejs/node/issues/14386
+// Fixes: https://github.com/nodejs/node/issues/14381
+
+class Response extends OutgoingMessage {
+  constructor() {
+    super({ method: 'GET', httpVersionMajor: 1, httpVersionMinor: 1 });
+  }
+
+  _implicitHeader() {}
+}
+
+const res = new Response();
+const ws = new Writable({
+  write: common.mustCall((chunk, encoding, callback) => {
+    assert(chunk.toString().match(/hello world/));
```

Comment:
the chunk contains a full HTTP response, which for brevity I did not want to include. `common.mustCall` it's also sufficient to validate the test on its own.